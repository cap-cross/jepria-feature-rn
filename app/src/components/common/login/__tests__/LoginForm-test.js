// // __tests__/LoginForm-test.js
// import 'react-native';
// // Note: test renderer must be required after react-native.
// import renderer from 'react-test-renderer';
// import React from 'react';

// import { reducer as formReducer } from 'redux-form';
// import { createStore, combineReducers } from 'redux';
// import { Provider } from 'react-redux';
// import { mount, shallow } from 'enzyme';

// import LoginForm from '../LoginForm';

// /* global beforeEach, describe, it,expect */

// describe('LoginForm Component', () => {
//   let responseState; // to test response state

//   beforeEach(() => {
//     const store = createStore(combineReducers({ form: formReducer }));
//     const returnState = (state) => {
//       responseState = state;
//       return state;
//     };

//     const subject = mount(
//       <Provider store={store}>
//         <LoginForm onSubmit={returnState} />
//       </Provider>,
//     );
//   });

//   it('renders correctly', () => {
//     // const tree = renderer.create(<LoginForm onSubmit={() => null} />).toJSON();
//     const tree = shallow(<LoginForm onSubmit={() => null} />);

//     expect(tree).toMatchSnapshot();
//   });
// });
