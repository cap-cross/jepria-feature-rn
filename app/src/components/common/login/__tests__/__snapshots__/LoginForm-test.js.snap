// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Container)
    style={
        Object {
            "alignSelf": "stretch",
          }
    }
    virtual={undefined}
>
    <Styled(Content)
        contentContainerStyle={
            Object {
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 200,
              }
        }
        virtual={undefined}
    >
        <Styled(Form)
            virtual={undefined}
        >
            <Field
                component={[Function]}
                name="username"
                placeholder="Имя"
                style={
                    Object {
                        "textAlign": "center",
                      }
                }
            />
            <Field
                component={[Function]}
                name="password"
                placeholder="Пароль"
                style={
                    Object {
                        "textAlign": "center",
                      }
                }
            />
            <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                    Object {
                        "alignItems": "center",
                      }
                }
            >
                <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                        Object {
                            "alignSelf": "stretch",
                            "backgroundColor": "red",
                            "color": "white",
                            "height": 40,
                            "lineHeight": 28,
                            "textAlign": "center",
                          }
                    }
                >
                    Войти
                </Text>
            </TouchableOpacity>
        </Styled(Form)>
        <ModalWaitBar />
    </Styled(Content)>
</Styled(Container)>,
  "nodes": Array [
    <Styled(Container)
      style={
            Object {
                  "alignSelf": "stretch",
                }
      }
      virtual={undefined}
>
      <Styled(Content)
            contentContainerStyle={
                  Object {
                        "flex": 1,
                        "justifyContent": "center",
                        "marginBottom": 200,
                      }
            }
            virtual={undefined}
      >
            <Styled(Form)
                  virtual={undefined}
            >
                  <Field
                        component={[Function]}
                        name="username"
                        placeholder="Имя"
                        style={
                              Object {
                                    "textAlign": "center",
                                  }
                        }
                  />
                  <Field
                        component={[Function]}
                        name="password"
                        placeholder="Пароль"
                        style={
                              Object {
                                    "textAlign": "center",
                                  }
                        }
                  />
                  <TouchableOpacity
                        activeOpacity={0.2}
                        onPress={[Function]}
                        style={
                              Object {
                                    "alignItems": "center",
                                  }
                        }
                  >
                        <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              style={
                                    Object {
                                          "alignSelf": "stretch",
                                          "backgroundColor": "red",
                                          "color": "white",
                                          "height": 40,
                                          "lineHeight": 28,
                                          "textAlign": "center",
                                        }
                              }
                        >
                              Войти
                        </Text>
                  </TouchableOpacity>
            </Styled(Form)>
            <ModalWaitBar />
      </Styled(Content)>
</Styled(Container)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <LoginForm
      onSubmit={[Function]}
/>,
    "_instance": LoginForm {
      "context": Object {},
      "getClass": [Function],
      "props": Object {
        "onSubmit": [Function],
      },
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <Styled(Container)
      style={
            Object {
                  "alignSelf": "stretch",
                }
      }
      virtual={undefined}
>
      <Styled(Content)
            contentContainerStyle={
                  Object {
                        "flex": 1,
                        "justifyContent": "center",
                        "marginBottom": 200,
                      }
            }
            virtual={undefined}
      >
            <Styled(Form)
                  virtual={undefined}
            >
                  <Field
                        component={[Function]}
                        name="username"
                        placeholder="Имя"
                        style={
                              Object {
                                    "textAlign": "center",
                                  }
                        }
                  />
                  <Field
                        component={[Function]}
                        name="password"
                        placeholder="Пароль"
                        style={
                              Object {
                                    "textAlign": "center",
                                  }
                        }
                  />
                  <TouchableOpacity
                        activeOpacity={0.2}
                        onPress={[Function]}
                        style={
                              Object {
                                    "alignItems": "center",
                                  }
                        }
                  >
                        <Text
                              accessible={true}
                              allowFontScaling={true}
                              disabled={false}
                              ellipsizeMode="tail"
                              style={
                                    Object {
                                          "alignSelf": "stretch",
                                          "backgroundColor": "red",
                                          "color": "white",
                                          "height": 40,
                                          "lineHeight": 28,
                                          "textAlign": "center",
                                        }
                              }
                        >
                              Войти
                        </Text>
                  </TouchableOpacity>
            </Styled(Form)>
            <ModalWaitBar />
      </Styled(Content)>
</Styled(Container)>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginForm
    onSubmit={[Function]}
/>,
}
`;
